### DISCUSSION TITLE
Active state of a pyscript State Variable

### DISCUSSION THREAD

**Discussion by jrlacharnay** [custom-components/pyscript#303] [Q&A]
Hi,

In my script I created a State Variable with `state.persist()` and `state.set()` (within `pyscript.` domain).

I'm also using an [Entities Card](https://www.home-assistant.io/lovelace/entities) to display the value of this state. This card can "have icons colored when entity is active".

So my question is, how can I define in pyscript when a state/entity is active or not?
If not possible directly, is there any workaround you see?

Thanks!
Jean-Rémy

**Comment by jrlacharnay** [comment]
In fact, there is one work-around, it's to use the [template-entity-row](https://github.com/thomasloven/lovelace-template-entity-row) card which allow to customize when an entity is highlighted, but in my case I'm using the [multiple-entities-row](https://github.com/benct/lovelace-multiple-entity-row) card, which is similar to the standard Entities Card on the "active" topic.

So, what I would like to know, is how HA consider an entity "active", and how to customize it within pyscript.

**Comment by jrlacharnay** [comment]
Replying again to myself :)
Another work-around is to use a binary_sensor template sensor + an attribute holding the "active" state:

```
template:
  binary_sensor:
    - name: "Wrapper sensor"
      state: "{{ state_attr('pyscript.my_entity', 'active') }}"

```
But still, I'm interested in a pyscript-only solution, that would prevent me to create yet another entity.

**Comment by qui3xote** [comment]
Quick guess here: HA is probably looking for the entity state to be 'on' in order to determine if it's active. So using state.set(entity, 'on') might work.

**Reply by jrlacharnay** [reply]
> Quick guess here: HA is probably looking for the entity state to be 'on' in order to determine if it's active. So using state.set(entity, 'on') might work.

No, it does not work. I think it works only for binary_sensor and maybe other types of entities, but not with pyscript entities. Thanks anyway :)

**Reply by qui3xote** [reply]
HA must need some definition of what is considered an 'active' state for each domain. That's getting deeper into the bowels of HA than I'm qualified for! BTW: how do you create an entity from pyscript? It's not in the documentation that I can see.

**Reply by jrlacharnay** [reply]
> BTW: how do you create an entity from pyscript? It's not in the documentation that I can see.

Sorry, I used the wrong term, I was refering to State Variables as documented [here](https://hacs-pyscript.readthedocs.io/en/stable/reference.html#state-variables). I'm creating the State with `state.persist()` and `state.set()`. If they are not "entities" for HA, that's maybe explaining why there is no active status.

Maybe [this pull request](https://github.com/custom-components/pyscript/pull/129) could help, but as far as I can see it is open since quite a long time now :(

Any help appreciated :)

**Reply by jrlacharnay** [reply]
Original post updated.
